### PR TITLE
Pre commit for golang

### PR DESCRIPTION
**Pull request by ntwyman**
Added the pre-commit hooks for golang and the dependencies for them (folks will need to `make deps` again when this lands).

Tidied up a couple of references to ProtoWebApp

Updated glide to match current versions to quiet a pre-commit warning.

### CODE REVIEW

File: bin/prereqs [hunk 1]
```diff
@@ -16,9 +16,8 @@ function has() {
 }
 
 has go "brew install go"
-has glide "brew install glide"
 has yarn "brew install yarn"
-has pre-commit "brew install pre-commit && pre-commit install"
```
Comment (tenshiemi) [inline bin/prereqs]: Just curious why this change?

Comment (ntwyman) [reply bin/prereqs]: I am trying to move stuff out of the bin/prereqs and bin/installdepencies.sh which tend to get out of sync and as much into the Makefile as possible.

As the check-ins for this ticket show it's possible for everything to work perfectly locally and then not build at all on CircleCI.
